src/main/java/elise/MaybeDate.java

```diff
@@ -0,0 +1,50 @@
+package elise;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Abstraction of a period represented by either String or date and time.
+ */
+public class MaybeDate {
```

Comment:
I like how you abstracted a class for the Date which can be either of type String or a LocalDate. Very creative 👍